addon/-private/functional-modifier-manager.js

```diff
@@ -0,0 +1,45 @@
+import { capabilities } from '@ember/modifier';
+
+const MODIFIER_ELEMENTS = new WeakMap();
+const MODIFIER_TEARDOWNS = new WeakMap();
+
+function teardown(modifier) {
+  const teardown = MODIFIER_TEARDOWNS.get(modifier);
+
+  if (teardown && typeof teardown === 'function') {
+    teardown();
+  }
+}
+
+function setup(modifier, element, args) {
+  const { positional, named } = args;
+  const teardown = modifier(element, positional, named);
+
+  MODIFIER_TEARDOWNS.set(modifier, teardown);
+}
+
+export default class FunctionalModifierManager {
+  capabilities = capabilities('3.13');
+
+  createModifier(factory) {
+    const { class: fn } = factory;
+
+    return (...args) => fn(...args);
```

Comment:
Could this return the `fn` directly, since it's not injecting anything anymore?

Reply:
I _think_ we do require this as the instance of the function is the instance used in the Weak Maps to track the element and the cleanup method.

I'll add a test that covers for this case in case someone in the future has the same thought you had!

Reply:
ah, that totally makes sense 👍 yeah a test or even just a comment would be good